pkg/controller/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -205,3 +226,65 @@ func intstrPtr(i int32) *intstr.IntOrString {
 	intstr := intstr.FromInt(int(i))
 	return &intstr
 }
+
+func expectReconcile(t *testing.T, requests chan reconcile.Request, errors chan error) {
+	t.Helper()
+
+	{
+		ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+		defer cancel()
+
+	LOOP0:
+		for range time.Tick(pollingInterval) {
+			select {
+			case recv := <-requests:
+				if recv == expectedRequest {
+					break LOOP0
```

Comment:
If you find what you wanted on the channel, why restart the loop?

Reply:
I'm breaking from the loop, not restarting.